redis/src/commands/json.rs

```diff
@@ -0,0 +1,231 @@
+// can't use rustfmt here because it screws up the file.
+#![cfg_attr(rustfmt, rustfmt_skip)]
+use crate::cmd::{cmd, Cmd};
+use crate::connection::ConnectionLike;
+use crate::pipeline::Pipeline;
+use crate::types::{FromRedisValue, RedisResult, ToRedisArgs};
+use crate::{RedisError, ErrorKind};
+
+#[cfg(feature = "cluster")]
+use crate::commands::ClusterPipeline;
+
+use serde::ser::Serialize;
+
+macro_rules! json_serialize_fail {
+	($err: expr) => {
+		Err(
+			RedisError::from((
+				ErrorKind::Serialize,
+				"Serialisation Error",
+				format!("{}", $err)
+			))
+		)
+	}
+}
```

Comment:
If you make an `impl From<serde_json::Error> for RedisError` you can just apply `?` to any `serde_json` functions within a method that returns `RedisResult`.

Reply:
I didn't think of that, but this looks so much better
```rust
Ok::<Cmd, RedisError>(
	cmd("JSON.ARRAPPEND")
		.arg_take(key)
		.arg_take(path)
		.arg_take(serde_json::to_string(value)?)
)
```

Reply:
Btw the `Ok::<Cmd, RedisError>` is necessary, without it the macro complains about not being able to infer types, which is strange but this is a fairly *clean-ish* workaround